### ISSUE TITLE
How to animate content of presented "popup viewController" during interactive/drag presentation

### ISSUE THREAD

**Issue by MichaelBechHansen** [LeoNatan/LNPopupController#211]
Great library...

Is there any way to animate the content of the presented viewController relative to a drag/interactive presentation.

Use case is for example to do like Apples Music.app which animates the album cover art from small to large when presenting.

**Comment by LeoNatan** [comment]
Hello,

I've been considering how to do this properly. In iOS 10, Apple finally gave the option to have proper interactive animations using `UIViewPropertyAnimator`. I am considering using that to allow interactive animations during presentation and dismiss.

Retarding the animation you describe, I wonder if it should not be a feature of the framework. The feature is tracked here:
https://github.com/LeoNatan/LNPopupController/issues/196

**Comment by MichaelBechHansen** [comment]
The album cover example from music.app was - well - just an example.
For the use case im considering using LNPopupController, I will have a more advanced animation.
This will most certainly not be suitable for the library, and I believe the actual animation done by the "popup viewcontroller" should not be forced by the library.
(I'm going for an interactive animation/presentation, where a cover art image is only one part - multiple labels and buttons will also be animating along with the interactive presentation)

With that being said, having a default implementation that mimics music.app would be great.

If you have ideas on how to implement this feature, I will be happy to invest some time in it (as opposed to rolling my own).

**Comment by LeoNatan** [comment]
I do have ideas and started work. Let's see how it goes.

**Comment by LeoNatan** [comment]
I've made good progress and barring any unforeseen issues, I will release a new major version with this feature. It's working quite nicely so far.

**Comment by LeoNatan** [comment]
You will use 
```
[self.transitionCoordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
		//animations
	} completion:nil];
```
to animate your content.

**Comment by MichaelBechHansen** [comment]
Perfect - that's actually exactly what I was looking for.
I'm looking forward to trying this out.

**Comment by LeoNatan** [comment]
Still have some issues preventing me from release.
I am using `UIViewPropertyAnimator`. When the user supplies a real animation to `animateAlongsideTransition:` or none, it works, but if the user provides an animation block with no actual animated content in it, I am seeing issues I am not yet sure how to solve.

**Comment by MichaelBechHansen** [comment]
Wouldn't that be a mis-use of the library?
And IMO should not necessarily be accounted for in the code... 
A "known issue" or FAQ-item describing it could be enough to steer people around it.

**Comment by LeoNatan** [comment]
Not necessarily. Sometimes you want to have logic inside your animation block which can lead to no-op, or cases where the value set in the animation block is equal to the existing value, again causing a no-op. It should be supported.

**Comment by LeoNatan** [comment]
This will be implemented in v3.0. I've started work on it, no hard dates.

**Comment by LeoNatan** [comment]
Closing in favor of #348.
